File: src/main/java/com/pedigree/print/PrintService.java
package com.pedigree.print;

import javafx.print.PrinterJob;
import javafx.scene.Node;

public class PrintService {
    public static void printNode(Node node) {
        if (node == null) return;
        PrinterJob job = PrinterJob.createPrinterJob();
        if (job != null && job.showPrintDialog(node.getScene().getWindow())) {
            boolean success = job.printPage(node);
            if (success) job.endJob();
        }
    }
}

File: src/main/kotlin/com/family/tree/print/PrintService.kt
package com.family.tree.print

import javafx.print.PrinterJob
import javafx.scene.Node

/**
 * Kotlin version of PrintService. Preserves Java static-style API via @JvmStatic.
 */
object PrintService {
    @JvmStatic
    fun printNode(node: Node?) {
        if (node == null) return
        val job = PrinterJob.createPrinterJob()
        if (job != null && job.showPrintDialog(node.scene.window)) {
            val success = job.printPage(node)
            if (success) job.endJob()
        }
    }
}
